File: Data Structures & Algorithms/merge-strings-alternately/submission-1.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int left = 0;
        int right = 0;
        String str = "";

        boolean l = true;
        while(left < word1.length() && right < word2.length()){
            if(l){
                str = str + word1.charAt(left);
                left++;
            }else {
                str = str + word2.charAt(right);
                right++;
            }
            l = !l;
        }
        if(left < word1.length()){
            str = str + word1.substring(left, word1.length());
        }else if(right < word2.length()){
            str = str + word2.substring(right, word2.length());
        }
        return str;
    }
}

File: Data Structures & Algorithms/edit-distance/submission-1.py
class Solution:
    def minDistance(self, word1: str, word2: str) -> int:
        n1 = len(word1)
        n2 = len(word2)
        dp = {}

        def operations(i: int, j: int):
            if  (i,j) in dp:
                return dp[(i,j)]
                
            if j == n2:
                return n1 - i
            elif i == n1:
                return n2 - j
            
            if word1[i] == word2[j]:
                dp[(i,j)] = operations(i+1, j+1)
                return dp[(i,j)]
            else:
                dp[(i,j)] = 1 + min(
                    operations(i+1, j),
                    operations(i+1, j+1),
                    operations(i, j+1)
                    )
                return dp[(i,j)]
        ops = operations(0, 0)
        return ops